parity/configuration.rs

```diff
@@ -424,9 +424,22 @@ impl Configuration {
 		self.args.flag_rpcapi.clone().unwrap_or(self.args.flag_jsonrpc_apis.clone())
 	}
 
-	pub fn rpc_cors(&self) -> Vec<String> {
+	pub fn rpc_cors(&self) -> Option<Vec<String>> {
 		let cors = self.args.flag_jsonrpc_cors.clone().or(self.args.flag_rpccorsdomain.clone());
-		cors.map_or_else(Vec::new, |c| c.split(',').map(|s| s.to_owned()).collect())
+		cors.map(|c| c.split(',').map(|s| s.to_owned()).collect())
+	}
+
+	pub fn rpc_hosts(&self) -> Option<Vec<String>> {
+		let hosts = self.args.flag_jsonrpc_hosts.split(',').collect::<Vec<&str>>();
```

Comment:
this algo will issue no error on `--jsonrpc-hosts=all,none` and silently assume `all`, which at best is non-obvious behaviour from the docs.

suggest matching specials prior to split.